### ISSUE TITLE
Change repetion_penalty calculation logic

### ISSUE THREAD

**Issue by qualiteger** [qualiteg/ChatStream#18]
Change the penalty logic from "multiplication" to "division" in accordance with the HuggingFace's Transformers calculation rules.

**Comment by qualiteger** [comment]
```
   if penalty is not None and past_tokens is not None:
        # Check if logits contain any abnormal values
        if torch.any(torch.isnan(logits)) or torch.any(torch.isinf(logits)):
            print("Logits contain NaN or Inf values. Replacing with very large / small numbers.")
            logits = torch.where(torch.isnan(logits), torch.full_like(logits, 0), logits)
            logits = torch.where(torch.isinf(logits) & (logits > 0),
                                 torch.full_like(logits, torch.finfo(logits.dtype).max),
                                 logits)
            logits = torch.where(torch.isinf(logits) & (logits < 0),
                                 torch.full_like(logits, torch.finfo(logits.dtype).min),
                                 logits)

        if not isinstance(penalty, (int, float)):
            raise ValueError(f"penalty should be a scalar value, but got {penalty}({type(penalty)})")

        if penalty_method == "multiplicative":
            if penalty != 1.0:
                for token in set(past_tokens):
                    logits[token] /= penalty
        elif penalty_method == "subtractive":
            for token_id in set(past_tokens):
                logits[token_id] -= penalty
        else:
            raise ValueError(f"Unknown penalty_method: {penalty_method}")
```

において、 `logits[token] *= penalty`  を `logits[token] /= penalty` に変更する。HuggingFace の Transformer のお作法にあわせる。

**Comment by qualiteger** [comment]
repetion_penalty が 1.05 の場合、

過去のトークンのロジット値を 1.05 で「割る」のがHuggingFace Transformerのお作法となる。「かける」ではない。

sampling関数であつかう logit は、モデルが出力したロジット列の 最後（最新）のロジットとなる。

ロジット列はつまりモデルへの入力文のトークン数ぶんあるが、
その最後のロジットから、何番目の語彙を選びだすかがサンプリングとなる。


たとえば、last_token_logits が 以下のような１次元テンソルの場合では、そのインデックスが語彙の種類を示しており、
値がロジット値をしめす。（これにソフトマックスをかけると確率になるので、ロジット値を確率のタマゴとみなす）

last_token_logits = torch.tensor([0.2, 0.4, 0.1, 0.3])

ここで過去に出現したトークン（語彙インデックス）は past_tokens にはいっているため、

```
for token in set(past_tokens):
                    logits[token] /= penalty
```

過去に past_tokens=[0,2]のときは語彙インデックス0、語彙インデックス2
つまりlast_token_logitsの要素１番目と３番目は

**Comment by qualiteger** [comment]
過去に出現しているため、penalty値でわって、値を小さくして出現確率をさげてやる。
